lib/zip/file.rb

```diff
@@ -415,15 +414,11 @@ def on_success_replace
     def create_tmpname
       dirname, basename = ::File.split(name)
       ::Dir::Tmpname.create(basename, dirname) do |tmpname|
-        opts = {perm: 0600, mode: ::File::CREAT | ::File::WRONLY | ::File::EXCL}
+        opts = {mode: ::File::CREAT | ::File::WRONLY | ::File::EXCL}
```

Comment:
At first I was worried that removing the `perm: 0600` bit of code was going to result in a security hole (i.e. the tempfile being created in a shared directory like `/tmp/` with permissions where another user could potentially read the file), *but* after much testing I validated that there is **not** a security risk! Woot! It became clear to me that because we're passing the basename to `Dir::Tmpname`, the tempfile is *not* being created in a shared space, unless the person using this gem specifically points `File::Zip.open` at a shared directory.

*Note:* this comment is just for posterity, in case a future dev has concerns about this same potential security flaw.